src/org/intellij/erlang/ErlangReferenceContributor.java

```diff
@@ -29,30 +30,39 @@
 
   @Override
   public void registerReferenceProviders(@NotNull PsiReferenceRegistrar registrar) {
-    register(registrar, Kind.RECORD,   "erlang", "is_record",  2, 1);
-    register(registrar, Kind.RECORD,   "erlang", "is_record",  3, 1);
-    register(registrar, Kind.MODULE,   "erlang", "spawn",      3, 0);
-    register(registrar, Kind.FUNCTION, "erlang", "spawn",      3, 1);
-    register(registrar, Kind.MODULE,   "erlang", "spawn_link", 3, 0);
-    register(registrar, Kind.FUNCTION, "erlang", "spawn_link", 3, 1);
+    registerRecords(registrar, "erlang", "is_record", 2, 1);
+    registerRecords(registrar, "erlang", "is_record", 3, 1);
+
+    registrar.registerReferenceProvider(psiElement(ErlangQAtom.class).with(
+                                          new ErlangPsiImplUtil.ErlangSpecParameter<PsiElement>()),
```

Comment:
This is not intuitive at all. ErlangSpecParameter doesn't give any idea what you're trying to match here.